### PR TITLE
Aftershock: Implement charge rifle

### PR DESCRIPTION
**Pull request by Pericles1848**
#### Summary
Mods "Implements a charge rifle, a new type of energy weapon into Aftershock"

#### Purpose of change
Implements charge rifle. This provides a new weapon in line with the setting of Aftershock. Further diversifies arsenal and adds interesting new option for players to use. Helps expand variety of energy weapons.

#### Describe the solution
This feature implements the charge rifle energy weapon. This weapon has two integrated weapon mods that increase the damage, and in the case of the second mod creates a small explosion, at the cost of energy consumption.

#### Describe alternatives you've considered
Implement other kinds of weapons to diversify Aftershock weaponry, not implement charge rifle,

#### Testing
Examined existing code for weapons that use integrated mods (KSG, Tavor, LeMat) and compared to new feature.

#### Additional context
Initially discussed on the Discord. Initially considered a 3rd integrated mod using EXPLOSION instead of SMALL_EXPLOSION and EMP, would have been unbalanced, but might work better as a final mod you can find separate from the gun.

### CODE REVIEW

File: data/mods/Aftershock/items/gun/combination.json [hunk 1]
```diff
@@ -15,5 +15,44 @@
...
     "blackpowder_tolerance": 60,
     "faults": [ "fault_gun_blackpowder", "fault_gun_dirt" ]
+  },
+  {
+    "id": "charge_rifle",
+    "looks_like": "ar_15",
+    "type": "GUN",
+    "reload_noise_volume": 10,
+    "name": { "str": "VER-3" },
+    "description": "Variable Energy Rifle Model 3, colloquially known as the 'charge rifle'. Designed by a consortium of private security forces, it is made to be used in multiple roles thanks to it's unique set of staged capacitors that can be used to increase the power of the gun exponetially. However this renders it incompatible with energy weapon mods.",
```
Comment (Night-Pryanik) [inline data/mods/Aftershock/items/gun/combination.json:25]:
```suggestion
    "description": "Variable Energy Rifle Model 3, colloquially known as the 'charge rifle'.  Designed by a consortium of private security forces, it is made to be used in multiple roles thanks to its unique set of staged capacitors that can be used to increase the power of the gun exponetially.  However this renders it incompatible with energy weapon mods.",
```

File: data/mods/Aftershock/items/gunmods/laser_gunmods.json [hunk 2]
```diff
@@ -9,6 +9,72 @@
     "range_modifier": -10,
     "ammo_to_fire_multiplier": 1.2
   },
+{
+	"id": "charge_capacitor_1",
+	"type": "GUNMOD",
+	"name": {
+		"str": "VER-3 stage 2 capacitor"
+	},
+	"description": "The second stage capacitor of the VER-3, increasing it's power and range to where it's comparable to a DMR, but losing the ability to fire full auto and doubling it's power consumption. It's irremovable ",
```
Comment (Night-Pryanik) [inline data/mods/Aftershock/items/gunmods/laser_gunmods.json:18]:
```suggestion
	"description": "The second stage capacitor of the VER-3, increasing its power and range to where it's comparable to a DMR, but losing the ability to fire full auto and doubling its power consumption.  It's irremovable.",
```

File: data/mods/Aftershock/items/gunmods/laser_gunmods.json [hunk 3]
```diff
@@ -9,6 +9,72 @@
...
+		"reload": 200
+	}
+  },
+  {
+ 	"id": "charge_capacitor_2",
+ 	"type": "GUNMOD",
+ 	"name": {
+ 		"str": "VER-3 stage 3 capacitor"
+ 	},
+ 	"description": "The third stage capacitor of the VER-3, further increasing it's power to where it creates a small explosion on impact, but quadruples it's power cost from the stage 2 capacitor, as well as increasing it's noise. It's irremovable ",
```
Comment (Night-Pryanik) [inline data/mods/Aftershock/items/gunmods/laser_gunmods.json:51]:
```suggestion
 	"description": "The third stage capacitor of the VER-3, further increasing its power to where it creates a small explosion on impact, but quadruples its power cost from the stage 2 capacitor, as well as increasing its noise.  It's irremovable.",
```

**Comment by Night-Pryanik** [comment]
You need to add it somewhere to spawn, otherwise it won't be available to obtain.

**Comment by Pericles1848** [comment]
> You need to add it somewhere to spawn, otherwise it won't be available to obtain.
I added it to the item group "afs_rare_guns", is that sufficient?

**Comment by Night-Pryanik** [comment]
Yeah, I suppose so.

File: data/json/items/armor/suits_protection.json [hunk 4]
```diff
@@ -846,6 +846,92 @@
...
+  {
+	"id": "officer_uniform",
+	"repairs_like": "trenchcoat",
+	"type": "ARMOR",
+	"name": {
+		"str": "Dress uniform"
+	},
+	"description": "A military officer's dress uniform from before the Cataclysm, complete with rank and ribbons. Just looking at it gives you an air of authority. Perhaps not the most practical military wear, but a very stylish option",
+	"weight": "1653 g",
+	"volume": "4750 ml",
```
Comment (casswedson) [inline data/json/items/armor/suits_protection.json:858]: Keep your PR small and focused, this is _very important_, it makes it easier to review, this uniform and the "combo cover" can be added in a different pull request, they are not aftershock content so they should not be in an aftershock pull request.

Comment (Pericles1848) [reply data/json/items/armor/suits_protection.json]: Yeah that was an accident sorry, I changed it now

File: data/mods/Aftershock/items/gun/combination.json [hunk 5]
```diff
@@ -15,5 +15,44 @@
...
     "blackpowder_tolerance": 60,
     "faults": [ "fault_gun_blackpowder", "fault_gun_dirt" ]
+  },
+  {
+    "id": "charge_rifle",
+    "looks_like": "ar_15",
+    "type": "GUN",
+    "reload_noise_volume": 10,
+    "name": { "str": "VER-3" },
+    "description": "Variable Energy Rifle Model 3, colloquially known as the 'charge rifle'.  Designed by a consortium of private security forces, it is made to be used in multiple roles thanks to its unique set of staged capacitors that can be used to increase the power of the gun exponetially.  However this renders it incompatible with energy weapon mods.",
```
Comment (casswedson) [inline data/mods/Aftershock/items/gun/combination.json:25]:
```suggestion
    "description": "Variable Energy Rifle Model 3, colloquially known as the 'charge rifle'.  Designed by a consortium of private security forces, it is made to be used in multiple roles thanks to its unique set of staged capacitors that can be used to increase the power of the gun exponentially.  However this renders it incompatible with energy weapon mods.",
```

**Comment by Pericles1848** [comment]
Sorry, I was trying to fix an issue with my master branch and must've closed this somehow, I didn't mean to.

File: data/mods/Aftershock/items/gun/combination.json [hunk 6]
```diff
@@ -15,5 +15,44 @@
...
     "faults": [ "fault_gun_blackpowder", "fault_gun_dirt" ]
+  },
+  {
+    "id": "charge_rifle",
+    "looks_like": "ar_15",
+    "type": "GUN",
+    "reload_noise_volume": 10,
+    "name": { "str": "VER-3" },
+    "description": "Variable Energy Rifle Model 3, colloquially known as the 'charge rifle'.  Designed by a consortium of private security forces, it is made to be used in multiple roles thanks to its unique set of staged capacitors that can be used to increase the power of the gun exponentially.  However this renders it incompatible with energy weapon mods.",
+    "weight": "5000 g",
```
Comment (John-Candlebury) [inline data/mods/Aftershock/items/gun/combination.json:26]:
```suggestion
    "weight": "10 kg",
```

File: data/mods/Aftershock/items/gun/combination.json [hunk 7]
```diff
@@ -15,5 +15,44 @@
...
+    "volume": "3 L",
```
Comment (John-Candlebury) [inline data/mods/Aftershock/items/gun/combination.json:27]:
```suggestion
    "volume": "12 L",
```

File: data/mods/Aftershock/items/gun/combination.json [hunk 8]
```diff
@@ -15,5 +15,44 @@
...
+    "material": [ "steel", "plastic" ],
+    "symbol": "(",
+    "color": "magenta",
+    "skill": "rifle",
+    "range": 20,
+    "ranged_damage": { "damage_type": "heat", "amount": 20, "armor_penetration": 4 },
+    "built_in_mods": [ "charge_capacitor_1", "charge_capacitor_2" ],
+    "dispersion": 150,
+    "durability": 8,
+    "modes": [ [ "DEFAULT", "semi-auto", 1 ], [ "AUTO", "auto", 3 ] ],
```
Comment (John-Candlebury) [inline data/mods/Aftershock/items/gun/combination.json:42]: Might be better to remove the AUTO mode in exchange for the ability to set power output.

File: data/mods/Aftershock/items/gun/combination.json [hunk 9]
```diff
@@ -15,5 +15,44 @@
...
+    "loudness": 8,
+    "ups_charges": 25,
```
Comment (John-Candlebury) [inline data/mods/Aftershock/items/gun/combination.json:44]:
```suggestion
    "ups_charges": 30,
```

File: data/mods/Aftershock/items/gun/combination.json (hunk 8 above)
Comment (Pericles1848) [reply data/mods/Aftershock/items/gun/combination.json:42]: Yeah, the ability to change its mode does sort of cover for what auto usually does (more damage for the cost of more ammo).

File: data/mods/Aftershock/items/gun/combination.json [hunk 10]
```diff
@@ -41,7 +41,7 @@
     "durability": 8,
     "modes": [ [ "DEFAULT", "semi-auto", 1 ], [ "AUTO", "auto", 3 ] ],
```
Comment (John-Candlebury) [inline data/mods/Aftershock/items/gun/combination.json:42]:
```suggestion
    "modes": [ "DEFAULT", "semi-auto", 1 ],
```

**Comment by Mom-Bun** [comment]
Speaking of, shouldn't this be using a magazine? If I recall that is important to energy weapons balance in AFS

**Comment by Maleclypse** [comment]
Conflicts can't be resolved in the web editor, I was going to try and fix them.

**Comment by Night-Pryanik** [comment]
@Pericles1848 could you please resolve conflicts?

**Comment by John-Candlebury** [comment]
After rethinking this concept, Im not sure its something that is actually necessary within the mod. Even if it we found it perfectly balanced, it would just be a laser rifle that you can also selectively fire as a large inefficient pistol.

And I think there is little point to carrying it vs an actually good laser rifle + a pistol. Aftershock lasers already use the same magazines, and its not like a pistol is particularly big and hard to fit in your inventory

**Pull request closed without merging**